recipes-core/tdx-enc-handler/tdx-enc-handler/tdx-enc-caam.sh

```diff
@@ -61,6 +61,20 @@ tdx_enc_key_gen() {
     fi
 }
 
+# backup original data in partition (if not encrypted)
+tdx_enc_backup_data() {
+    mkdir -p "${TDX_ENC_STORAGE_MOUNTPOINT}"
+    if mount -t ext4 ${TDX_ENC_STORAGE_LOCATION} "${TDX_ENC_STORAGE_MOUNTPOINT}"; then
```

Comment:
I think we should have a variable (i.e. `TDX_ENC_KEEP_DATA`) so the user can enable or disable this behaviour, since some users might just want to ignore the content of the partition and format it.

This variable might be `0` by default, but users could enable this behaviour by setting it to `1`.

Reply:
Already on it, my most recent changes locally has `TDX_ENC_PRESERVE_DATA` but that has a bit of a ripple effect in the layer as more than 1 file needs to change for that.